engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/FetchCachedResultsActor.scala

```diff
@@ -2,37 +2,35 @@ package cromwell.engine.workflow.lifecycle.execution.callcaching
 
 import akka.actor.{Actor, ActorLogging, ActorRef, Props}
 import cromwell.Simpletons._
-import cromwell.core.JobOutputs
-import cromwell.core.simpleton.WdlValueBuilder
+import cromwell.core.simpleton.WdlValueSimpleton
 import cromwell.engine.workflow.lifecycle.execution.callcaching.EngineJobHashingActor.CacheHit
 import cromwell.engine.workflow.lifecycle.execution.callcaching.FetchCachedResultsActor.{CachedOutputLookupFailed, CachedOutputLookupSucceeded}
-import cromwell.services.SingletonServicesStore
-import wdl4s.TaskOutput
 
 import scala.concurrent.ExecutionContext
 import scala.util.{Failure, Success}
 
 object FetchCachedResultsActor {
-  def props(cacheHit: CacheHit, taskOutputs: Seq[TaskOutput], replyTo: ActorRef, callCache: CallCache): Props =
-    Props(new FetchCachedResultsActor(cacheHit, taskOutputs, replyTo, callCache))
+  def props(cacheHit: CacheHit, replyTo: ActorRef, callCache: CallCache): Props =
+    Props(new FetchCachedResultsActor(cacheHit, replyTo, callCache))
 
-  sealed trait CachedOutputLookupResponse
-  case class CachedOutputLookupFailed(metaInfoId: MetaInfoId, failure: Throwable) extends CachedOutputLookupResponse
-  case class CachedOutputLookupSucceeded(jobOutputs: JobOutputs, cacheHit: CacheHit) extends CachedOutputLookupResponse
+  sealed trait CachedResultResponse
+  case class CachedOutputLookupFailed(metaInfoId: MetaInfoId, failure: Throwable) extends CachedResultResponse
+  case class CachedOutputLookupSucceeded(simpletons: Seq[WdlValueSimpleton], callOutputFiles: Map[String,String], returnCode: Option[Int], cacheHit: CacheHit) extends CachedResultResponse
 }
 
 
-class FetchCachedResultsActor(cacheHit: CacheHit, taskOutputs: Seq[TaskOutput], replyTo: ActorRef, callCache: CallCache)
+class FetchCachedResultsActor(cacheHit: CacheHit, replyTo: ActorRef, callCache: CallCache)
   extends Actor with ActorLogging {
+
   {
     implicit val ec: ExecutionContext = context.dispatcher
     val cacheResultId = cacheHit.cacheResultId
 
     callCache.fetchCachedResult(cacheResultId) onComplete {
       case Success(Some(result)) =>
         val simpletons = result.resultSimpletons map toSimpleton
-        val jobOutputs = WdlValueBuilder.toJobOutputs(taskOutputs, simpletons)
-        replyTo ! CachedOutputLookupSucceeded(jobOutputs, cacheHit)
+        val jobOutputFiles = result.jobDetritus map { jobDetritusEntry => jobDetritusEntry.jobDetritusKey -> jobDetritusEntry.jobDetritusValue }
```

Comment:
I would rename `jobOutputFiies` to be consistent with the detritus theme.

And then let @mcovarr and @geoffjentry fight about whether detritus is great or terrible!